### PR TITLE
Updated umccrise step to take germline as input rather than its own step

### PR DESCRIPTION
**Pull request by alexiswl**
Relates to https://github.com/umccr/infrastructure/pull/284

dragen somatic now runs dragen somatic + germline pipeline

umccrise step now takes dragen germline directory output as input rather than running germline as its own step.

### CODE REVIEW

File: data_processors/pipeline/orchestration/tests/test_umccrise_step.py
Comment (victorskl) [inline data_processors/pipeline/orchestration/tests/test_umccrise_step.py:213]: Do you happen to run umccrise; based on output of this somatic+germline workflow run output? If so, could you pointer wfr id pls?

File: data_processors/pipeline/orchestration/umccrise_step.py
Comment (victorskl) [inline data_processors/pipeline/orchestration/umccrise_step.py:82]: Looks like, input structure has changed a bit. This job struct will have to have a binding contract with the job trigger Lambda counterpart.

https://github.com/umccr/data-portal-apis/blob/344a7936503ab8f9ae276ce8da5c09eec5a18dc9/data_processors/pipeline/lambdas/umccrise.py#L62-L90

Perhaps, this has to stem from input template JSON changes; discussed upstream at infra repo. ditto

https://github.com/umccr/infrastructure/pull/284#pullrequestreview-1327229613

File: data_processors/pipeline/tools/liborca.py
Comment (victorskl) [inline data_processors/pipeline/tools/liborca.py:114]: Unit test itself for this function in `test_liborca.py` would be nice.

**Review by victorskl — COMMENTED** [review]
Hey Alexis; this change input/output struct a bit. No worries. 

So. I reckon, I am expecting those input/output template JSON struct updates has to come first at infra repo? 

https://github.com/umccr/infrastructure/pull/284

Then. We can code along those template JSON changes accordingly over here. Both on; event contract between --
- Job Producer (i.e. step modules in orchestration package)
- Job Consumer (i.e. triggering Lambda that subscribe to the queue)

I am happy to hold the fort on this Portal side; so that you can concentrate finalising up 3.9.3 somatic+germline workflow and, follow up umccrise changes need in there.

Happy to chat; if any. Still holding this PR for now.

**Comment by victorskl** [comment]
@alexiswl Alexis; seeing [Slack thread](https://umccr.slack.com/archives/C025SCF86/p1678139733455829), I'm happy to take over the branch continue dev work here. So. Please no worries on this side and tackling comments above.

Please just help me drop in those updated JSON template accordingly in infra repo once you finalised. I can continue from then on.

File: data_processors/pipeline/orchestration/umccrise_step.py
Comment (alexiswl) [reply data_processors/pipeline/orchestration/umccrise_step.py:82]: Updated the event payload in the documentation for umccrise, should be able to resolve this conversation

File: data_processors/pipeline/orchestration/tests/test_umccrise_step.py
Comment (alexiswl) [reply data_processors/pipeline/orchestration/tests/test_umccrise_step.py:213]: Doing this now

File: data_processors/pipeline/lambdas/umccrise.py
Comment (victorskl) [inline data_processors/pipeline/lambdas/umccrise.py:101]: `workflow_input['output_file_prefix']` is gone? I guess, yes. Just noting. I will tackle it.

File: data_processors/pipeline/tools/liborca.py
Comment (victorskl) [reply data_processors/pipeline/tools/liborca.py:114]: I will tackle this

**Review by victorskl — COMMENTED** [review]
Thanks Alexis for updates. Me going through _snail-paced_ review here as this is quite critical changes; automation-wise.

Minor comments to attend and unit tests; for that, pls allow me to do it myself this point onwards. So. I will take over the branch; if you don't mind.

I will peek into DEV for new umccrise run to observe; when completed.

File: data_processors/pipeline/lambdas/umccrise.py
Comment (alexiswl) [reply data_processors/pipeline/lambdas/umccrise.py:101]: Ah yes, output file prefix is now no longer required.

**Comment by alexiswl** [comment]
Launched  wfr.f93fcc7d267b43b0a6c5856ccf3e0134 based on outputs of wfr.5616c72c82e442f78f0f9f0d6441219e

File: data_processors/pipeline/orchestration/tests/test_umccrise_step.py
Comment (alexiswl) [reply data_processors/pipeline/orchestration/tests/test_umccrise_step.py:213]: Workflow run id is wfr.f93fcc7d267b43b0a6c5856ccf3e0134

**Comment by alexiswl** [comment]
Thanks @victorskl, workflow failed, from the task stderr looks like I need to be more careful with the names of the sample inputs

**Comment by alexiswl** [comment]
Relaunched under `wfr.3bb2f749f96f44eaa6e3c67ec43d6111`, and updated dragen_normal_id and dragen_tumor_id parameters to be the sample name (not the library name)

**Comment by alexiswl** [comment]
Have also just updated the umccrise json template in infrastructure accordingly [umccr/infrastructure@`9df1464` (#284)](https://github.com/umccr/infrastructure/pull/284/commits/9df14640ec43eab7a07d6d7ff6cf7e5498d1474b)

File: data_processors/pipeline/lambdas/umccrise.py
Comment (alexiswl) [inline data_processors/pipeline/lambdas/umccrise.py:89]: We also need the sample ids to go in as inputs. For the umccrise tool these are dragen_tumor_id and dragen_normal_id, these should match the sample name used (RGSM in fastq list rows). UMCCRise requires these match the values in the VCF

Comment (victorskl) [reply data_processors/pipeline/lambdas/umccrise.py:89]: Sure; got it. I will update it appropriate rgsm/sample_id mapping there, yes.

Comment (victorskl) [reply data_processors/pipeline/lambdas/umccrise.py:89]: Looks like `wfr.3bb2f749f96f44eaa6e3c67ec43d6111` succeeded. I will push my changes in tick, next. Let us review each other, ok.

Comment (victorskl) [reply data_processors/pipeline/lambdas/umccrise.py:91]: @alexiswl Some improved made with https://github.com/umccr/data-portal-apis/commit/41b70ff28a97f4b7c64b6e309d1784e7a6e9aa4f Please kindly check it at your end.

**Comment by victorskl** [comment]
@williamputraintan T/N payload has changed with this PR. So T/N `/pairing` endpoint response will be changed as well. Something to follow up on UI T/N trigger. After this get merged.

File: data_processors/pipeline/lambdas/umccrise.py
Comment (victorskl) [reply data_processors/pipeline/lambdas/umccrise.py:91]: @williamputraintan T/N payload has changed with this PR. So T/N `/pairing` endpoint response will be changed as well. You can peek the diff here. Something to follow up on UI T/N trigger. After this get merged.

Comment (victorskl) [reply data_processors/pipeline/lambdas/umccrise.py:91]: @reisingerf Added you as second pair of eye for this, pls.

**Review by reisingerf — APPROVED** [review]
LGTM

**Review by victorskl — APPROVED** [review]
@alexiswl If you have no further comments; let us get merged in this. Then, we shall use new Portal UI feature T/N trigger to give it a test run in DEV for these changes. 🙂

File: data_processors/pipeline/lambdas/umccrise.py
Comment (alexiswl) [reply data_processors/pipeline/lambdas/umccrise.py:91]: Amazing, thanks @victorskl

**Comment by victorskl** [comment]
Thanks all, merging...

**Pull request merged**
